### PR TITLE
[docs] Fix broken API reference link

### PR DESCRIPTION
**Pull request by oliviertassinari**
Fix https://app.ahrefs.com/site-audit/3992793/31/data-explorer?columns=pageRating%2Curl%2Ctraffic%2ChttpCode%2CcontentType%2Cdepth%2CincomingAllLinks&filterId=624c70b6a74a7ac140893dd4714b6d3f&issueId=c64d3b5a-d0f4-11e7-8ed1-001e67ed4656&sorting=-pageRating

It's used in locations like: https://stackoverflow.com/questions/74389125/mui-datepicker-how-can-i-change-yearpicker-styling

### CODE REVIEW

**Comment by alexfauquette** [comment]
On monday, I will take some time to list pages in v5 and pages in v6 to verify we do not have more redirect to do

**Comment by oliviertassinari** [comment]
@alexfauquette Ah right, I didn't think about this, there could be more. You can find them at https://search.google.com/search-console/index/drilldown?resource_id=sc-domain%3Amui.com&item_key=CAMYDSAC: 

Screenshot 2023-04-01 at 13 07 37

I'm not sure how important it is, but I fixed them in the past, in the doubts.

**Review by alexfauquette — APPROVED** [review]

File: docs/public/_redirects [hunk 1]
```diff
@@ -44,5 +44,17 @@
...
 /x/whats-new/ https://mui.com/blog/mui-x-v6/ 301
+/x/api/date-pickers/calendar-picker/ /x/api/date-pickers/date-calendar/ 301
+/x/api/date-pickers/calendar-picker-skeleton/ /x/api/date-pickers/day-calendar-skeleton/ 301
+/x/api/date-pickers/month-picker/ /x/api/date-pickers/month-calendar/ 301
+/x/api/date-pickers/year-picker/ /x/api/date-pickers/year-calendar/ 301
+/x/react-data-grid/editing-legacy/ https://https://v5.mui.com/x/react-data-grid/editing-legacy/ 301
+/x/api/date-pickers/desktop-next-date-picker/
+/x/api/date-pickers/desktop-date-picker/ 301
+/x/api/date-pickers/next-date-picker/ /x/api/date-pickers/date-picker/ 301
+/x/react-date-pickers/calendar/
```
Comment (oliviertassinari) [inline docs/public/_redirects:55]: This lin is strange, are you sure about it? I would expect `in out 301`.

there are a few others as well.

Comment (alexfauquette) [reply docs/public/_redirects:55]: Yes, I don't  know why those spaces get replaced by line breaks. Fixed in #8524

Comment (oliviertassinari) [reply docs/public/_redirects:55]: Perfect, thanks. 

We also have https://app.ahrefs.com/v2-site-explorer/broken-backlinks?anchorRules=&domainNameRules=&followType=all&grouping=one-per-domain&limit=50&mode=prefix&offset=0&refPageTitleRules=&refPageUrlRules=&sort=traffic&sortDirection=desc&surroundingRules=&target=https%3A%2F%2Fmui.com%2F&targetUrlRules= at our disposal to find links to our docs that are broken.

File: docs/public/_redirects [hunk 2]
```diff
@@ -44,5 +44,17 @@
 /:lang/x/* /x/:splat 301
 # 2023
 /x/whats-new/ https://mui.com/blog/mui-x-v6/ 301
+/x/api/date-pickers/calendar-picker/ /x/api/date-pickers/date-calendar/ 301
+/x/api/date-pickers/calendar-picker-skeleton/ /x/api/date-pickers/day-calendar-skeleton/ 301
+/x/api/date-pickers/month-picker/ /x/api/date-pickers/month-calendar/ 301
+/x/api/date-pickers/year-picker/ /x/api/date-pickers/year-calendar/ 301
+/x/react-data-grid/editing-legacy/ https://https://v5.mui.com/x/react-data-grid/editing-legacy/ 301
```
Comment (oliviertassinari) [inline docs/public/_redirects:51]: Oops `https://https://` 🙃. Fixed in 6e3ce20b130be51741b18444ba63eff64a7c3930

**Pull request merged**